Sluggo/View Controllers/TicketDetailTableViewController.swift

```diff
@@ -0,0 +1,231 @@
+        currentMember = ticket?.assigned_user ?? nil
+        dueDatePicker.date = ticket?.due_date ?? Date()
+        dueDatePicker.isHidden = (ticket?.due_date == nil)
+        dueDateLabel.isHidden = (ticket?.due_date != nil)
+        dueDateSwitch.isEnabled = (ticket?.due_date != nil)
+        dueDatePicker.isEnabled = dueDateSwitch.isEnabled
+        assignedField.isEnabled = true
+        navBar.title = self.ticket != nil ? "Selected Ticket" : "Create a Ticket"
+        setEditMode(self.ticket == nil)
+        createUserPicker()
+        createStatusPicker()
+        rightButton.title = (self.ticket != nil) ? "Edit" : "Done"
+    }
+
+    func setEditMode(_ editing: Bool) {
+        editingTicket = editing
+        ticketTitle.isUserInteractionEnabled = editing
+        ticketDescription.isUserInteractionEnabled = editing
+        dueDatePicker.isEnabled = true
+        dueDateSwitch.isEnabled = true
+        dueDateSwitch.isOn = (ticket?.due_date != nil)
+        dueDatePicker.isUserInteractionEnabled = editing
+        dueDateSwitch.isUserInteractionEnabled = editing
+        assignedField.isUserInteractionEnabled = editing
+        statusField.isUserInteractionEnabled = editing
+        dueDateSwitch.isHidden = !editing
+        dueDatePicker.isHidden = (ticket?.due_date == nil && !editing)
+        dueDateLabel.isHidden = (ticket?.due_date != nil || editing)
+    }
+
+    func doSave() {
+        let title = ticketTitle.text ?? "Default Title"
+        let description = ticketDescription.text
+        let date = dueDateSwitch.isOn ? dueDatePicker.date : nil
+        let member = currentMember?.id
+        let status = currentStatus?.id
+
+        if editingTicket {
+            ticket!.title = title
+            ticket!.description = description
+            ticket!.due_date = date
+            ticket!.assigned_user = currentMember
+            ticket!.status = currentStatus
+
+            let manager = TicketManager(identity)
+
+            manager.updateTicket(ticket: ticket!) { result in
+                self.processResult(result: result) { _ in
+                    DispatchQueue.main.async {
+                        self.setEditMode(false)
+                        NotificationCenter.default.post(name: .refreshTrigger, object: self)
+                    }
+                }
+            }
+        } else {
+            let ticket = WriteTicketRecord(tag_list: [],
+                                           assigned_user: member,
+                                           status: status,
+                                           title: title,
+                                           description: description,
+                                           due_date: date)
+            let manager = TicketManager(identity)
+            manager.makeTicket(ticket: ticket) { result in
+                self.processResult(result: result) { _ in
+                    DispatchQueue.main.async {
```

Comment:
processResult _should_ dispatch to the main thread by itself

Reply:
I think that was just blind copying from the old version, will be removed (or it can be removed after the merge in next PR)